ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -170,6 +171,11 @@ internal class PaywallViewModelImpl(
                 val customerInfo = purchases.awaitRestore()
                 Logger.i("Restore purchases successful: $customerInfo")
                 listener?.onRestoreCompleted(customerInfo)
+                shouldDisplayBlock?.let {
+                    if (!it(customerInfo)) {
+                        options.dismissRequest()
+                    }
+                }
```

Comment:
Nit: maybe add some line breaks in this method? There's 5 different completely things being done, so I'd separate them a bit.